src/modules/crud.js

```diff
@@ -0,0 +1,109 @@
+import Task from './task.js';
+import addEventListeners from './utils.js';
+
+export default class ToDo {
+  constructor() {
+    this.todoList = [];
+  }
+
+  renderTask(task) {
+    // render task to the page
+    // ....here's the code to render the li element
+    const liElement = document.createElement('li');
+    liElement.classList.add('draggable-item');
+    liElement.setAttribute('value', task.index);
+    liElement.setAttribute('draggable', 'true');
+
+    liElement.innerHTML = `
+        <input class="completed-checkbox" type="checkbox">
+        <input class="task-description" type="text" name="${task.index}" value="${task.description}">
+
+        <i class="list-icon trash-icon"><ion-icon name="trash-outline"></ion-icon></i>
+        <i class="list-icon drag-icon"><ion-icon name="ellipsis-vertical-sharp"></ion-icon></i>
+        `;
+
+    // after rendering, add event listeners to the task
+    addEventListeners(liElement, this);
+
+    // set checkbox state
+    liElement.querySelector('.completed-checkbox').checked = task.completed;
+    // Set line-through style to the task description if the checkbox is checked
+    liElement.style.textDecoration = task.completed ? 'line-through' : 'none';
+
+    // finally append the liElement to the todo-list
+    document.querySelector('#draggable_list').appendChild(liElement);
+  }
+
+  updateIndexes() {
+    // update list index of the remaining tasks
+    document.querySelectorAll('.draggable-item')
+      .forEach((item, index) => {
+        item.setAttribute('value', index + 1);
+        item.querySelector('.task-description').setAttribute('name', index + 1);
+      });
+
+    // Must have its own forEach loop because of the way the DOM is structured
+    this.todoList.forEach((task, index) => {
+      task.index = index + 1;
+    });
+  }
+
+  addTask(description) {
+    const task = new Task(this.todoList.length + 1, description);
+
+    // add task to the array list
+    this.todoList.push(task);
+
+    // update local storage
+    localStorage.setItem('todo-list', JSON.stringify(this.todoList));
+
+    // render task into the page
+    this.renderTask(task);
+  }
+
+  removeTask(liElement) {
+    // get the index of the task to be removed
+    const taskIndex = liElement.value - 1;
+
+    // remove task from the array list
+    this.todoList.splice(taskIndex, 1);
+    // remove task from the page
+    liElement.remove();
+    // update list index of the remaining tasks
+    this.updateIndexes();
+    // update local storage
+    localStorage.setItem('todo-list', JSON.stringify(this.todoList));
+  }
```

Comment:
- [ ] Hello @ZikaZaki , the delete task method is not beign called, therefore I am not able to delete any task, kindly fix this

Reply:
Hi @VICTOR-CHUKWU,
Regarding the remove method, it is working perfectly and I have tested it a million times. Please check it again and check your `webpack` or `vs code`.

Reply:
Hey @VICTOR-CHUKWU  you are mistaken, I have rechecked it again and it is working perfectly. And to prove that I am 100% right I have just published the project on GitHub-Pages. Here's the link to the live demo: https://zikazaki.github.io/todo-list/dist/index.html. Please do not make this mistake again, because some novice students/programmers will get frustrated and even more will mess up their projects.